contracts/workflows/RoyaltyWorkflows.sol

```diff
@@ -0,0 +1,232 @@
+    /// and claims revenue on that snapshot for each specified currency token.
+    /// @param ancestorIpId The address of the ancestor IP.
+    /// @param currencyTokens The addresses of the currency (revenue) tokens to claim (each address must be unique).
+    /// @param royaltyClaimDetails The details of the royalty claim from child IPs,
+    /// see {IRoyaltyWorkflows-RoyaltyClaimDetails}.
+    /// @return snapshotId The ID of the snapshot taken.
+    /// @return amountsClaimed The amount of revenue claimed for each currency token.
+    function transferToVaultAndSnapshotAndClaimByTokenBatch(
+        address ancestorIpId,
+        address[] calldata currencyTokens,
+        RoyaltyClaimDetails[] calldata royaltyClaimDetails
+    ) external returns (uint256 snapshotId, uint256[] memory amountsClaimed) {
+        (snapshotId, amountsClaimed, ) = _transferToVaultAndSnapshotAndClaimByTokenBatch(
+            ancestorIpId,
+            currencyTokens,
+            royaltyClaimDetails
+        );
+    }
+
+    /// @notice Transfers royalties to the ancestor IP's royalty vault, takes a snapshot, claims revenue for each
+    /// specified currency token both on the new snapshot and on each specified unclaimed snapshots.
+    /// @param ancestorIpId The address of the ancestor IP.
+    /// @param currencyTokens The addresses of the currency (revenue) tokens to claim (each address must be unique).
+    /// @param unclaimedSnapshotIds The IDs of unclaimed snapshots to include in the claim.
+    /// @param royaltyClaimDetails The details of the royalty claim from child IPs,
+    /// see {IRoyaltyWorkflows-RoyaltyClaimDetails}.
+    /// @return snapshotId The ID of the snapshot taken.
+    /// @return amountsClaimed The amounts of revenue claimed for each currency token.
+    function transferToVaultAndSnapshotAndClaimBySnapshotBatch(
+        address ancestorIpId,
+        address[] calldata currencyTokens,
+        uint256[] calldata unclaimedSnapshotIds,
+        RoyaltyClaimDetails[] calldata royaltyClaimDetails
+    ) external returns (uint256 snapshotId, uint256[] memory amountsClaimed) {
+        // Claims revenue for each specified currency token from the latest snapshot
+        IIpRoyaltyVault ancestorIpRoyaltyVault;
+        (snapshotId, amountsClaimed, ancestorIpRoyaltyVault) = _transferToVaultAndSnapshotAndClaimByTokenBatch(
+            ancestorIpId,
+            currencyTokens,
+            royaltyClaimDetails
+        );
+
+        // Claims revenue for each specified currency token from the unclaimed snapshots
+        uint256 length = royaltyClaimDetails.length;
+        for (uint256 i = 0; i < length; i++) {
+            amountsClaimed[i] += ancestorIpRoyaltyVault.claimRevenueOnBehalfBySnapshotBatch({
+                snapshotIds: unclaimedSnapshotIds,
+                token: royaltyClaimDetails[i].currencyToken,
+                claimer: address(ancestorIpRoyaltyVault)
```

Comment:
When doing the final call that will take the money from a vault to a royalty token holder the `claimer` can be any address in the world that holds royalty tokens. So `claimer` that we added in last protocol PR on the functions `claimRevenueOnBehalfBySnapshotBatch` and `claimRevenueOnBehalfByTokenBatch` are input from the should be passed in to the SPG functions as the SPG has no way to know what this address is

Reply:
I see, will add an extra `claimer` param to these functions 🫡